### PR TITLE
Fix broken examples

### LINKED ISSUE #143: Broken Examples
[This commit](https://github.com/thephpleague/tactician/commit/aed75b326c56834f9d8b8bbdbf37dbe951e238bd) removed the dev-dep `league/container`, which was used by the examples.

See the evidence below:

```bash
$ php examples/1-beginner-standard-usage.php 
PHP Fatal error:  Uncaught Error: Class "League\Container\Container" not found in /Users/flavio/Documents/open-source/tactician/examples/1-beginner-standard-usage.php:26
Stack trace:
#0 {main}
  thrown in /Users/flavio/Documents/open-source/tactician/examples/1-beginner-standard-usage.php on line 26
```

The solution is to either refactor the examples or add back the dev-dep.

I'm happy to provide a PR fixing this as long as the direction is pointed out.

PS: This only affects the master branch afaik.

### PR DESCRIPTION
**Pull request by flavioheleno**
This PR fixes #143 and adds example support to CI to ensure examples are kept valid with changes to the package source.

### CODE REVIEW

File: tests/ExamplesTest.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace League\Tactician\Tests;
+
+use PHPUnit\Framework\TestCase;
+
+class ExamplesTest extends TestCase
+{
+
+    /**
+     * @dataProvider exampleFiles
```
Comment (rosstuck) [inline tests/ExamplesTest.php:13]:
```suggestion
     * @dataProvider exampleFiles
     * @param string[] $expect
```

File: tests/ExamplesTest.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            ),
+            $output,
+            $exitCode
+        );
+
+        $this->assertSame(0, $exitCode);
+        $this->assertSame($expect, $output);
+    }
+
+    public function exampleFiles() : array
```
Comment (rosstuck) [inline tests/ExamplesTest.php:36]:
```suggestion
    /** @return string[]
    public function exampleFiles() : array
```

File: tests/ExamplesTest.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                    '-f',
+                    escapeshellarg(__DIR__ . "/../examples/{$file}")
+                ]
+            ),
+            $output,
+            $exitCode
+        );
+
+        $this->assertSame(0, $exitCode);
+        $this->assertSame($expect, $output);
```
Comment (rosstuck) [inline tests/ExamplesTest.php:33]: These might need to be self:: instead of $this-> calls.

**Review by rosstuck — APPROVED** [review]
Couple of PHPStan fixes to clarify the types and it's good to go! :+1:

**Comment by flavioheleno** [comment]
Should be good to go now @rosstuck :-D

**Comment by rosstuck** [comment]
Lovely, thanks a ton! :D

**Pull request merged**